pkg/apis/core/types.go

```diff
@@ -5394,3 +5403,23 @@ type TopologySpreadConstraint struct {
 	// +optional
 	LabelSelector *metav1.LabelSelector
 }
+
+// PortStatus represents the error condition of a service port
+type PortStatus struct {
+	// Port is the port number of the service port of which status is recorded here
+	Port int32
+	// Protocol is the protocol of the service port of which status is recorded here
+	Protocol Protocol
```

Comment:
Will this default to TCP like other instances of Protocol or is it just required?  Given that this is not a user-oriented API it's fine to not default (I just wanted to ask)

Reply:
It is just required. Actually, I have not thought about defining a default value for it.